### DISCUSSION TITLE
Chatflow calling an Internal Chatflow: How to override Session ID for Langfuse?

### DISCUSSION THREAD

**Discussion by dkindlund** [FlowiseAI/Flowise#1587] [General]
So I have a main chatflow that's internally calling *another* chatflow.  Both chatflows are logging to Langfuse.  Because each chatflow invocation has a unique "sessionId", Langfuse doesn't correlate the traces for this activity together in a single session.

My question is: Is it possible to take the $flow.chatId from the *first* chatflow and pass it into the internal *second* chatflow as an *override*?  That way, when Langfuse reports the activity, it shows both chatflow invocations tied to the same session.

Let me know your thoughts here, @HenryHengZJ .  Thanks!

**Comment by HenryHengZJ** [comment]
@dkindlund not sure if this helps, but just merged a PR which outputs the chatId along with the text response. Otherwise, I'd suggest provide the `chatId` in the API body request, so it can override the automatically generated one

**Reply by dkindlund** [reply]
Hey @HenryHengZJ , just to confirm, so if I provide a `chatId` in the API body request, Flowise will *automatically* use that one instead of generating a new `chatId` when executing a chatflow?

**Reply by HenryHengZJ** [reply]
yep: 
![image](https://github.com/FlowiseAI/Flowise/assets/26460777/a276fee6-d7fe-44ca-b70c-07aed1f9a81e)

**Reply by dkindlund** [reply]
Wow, that's awesome -- thanks so much, @HenryHengZJ!
